scripts/run_backref_cache_search_experiment.py

```
@@ -0,0 +1,579 @@
+                            stderr=subprocess.STDOUT, check=False)
+    if result.returncode != 0:
+        raise SystemExit("free-memory prerequisite unavailable: vm_stat failed")
+    try:
+        return parse_vm_stat_available_bytes(result.stdout)
+    except ValueError as error:
+        raise SystemExit(
+            f"free-memory prerequisite unavailable: {error}") from error
+
+
+def diagnostic(argv: list[str]) -> str:
+    return subprocess.run(argv, cwd=ROOT, text=True, stdout=subprocess.PIPE,
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)